### ISSUE TITLE
On event as array in configs

### ISSUE THREAD

**Issue by githubjeka** [yiisoft/yii2#14985]
REL https://github.com/yiisoft/yii2/issues/1988

### What steps will reproduce the problem?
HumHub use config:
```php
'db' => [
            'class' => 'yii\db\Connection',
            'dsn' => 'mysql:host=localhost;dbname=humhub',
            'username' => '',
            'password' => '',
            'charset' => 'utf8',
            'enableSchemaCache' => true,
            'on afterOpen' => ['humhub\libs\Helpers', 'SqlMode'],
],
```

I use config:

```php
'db' => [
            'class' => 'yii\db\Connection',
            'dsn' => 'mysql:host=localhost;dbname=humhub',
            'username' => '',
            'password' => '',
            'charset' => 'utf8',
            'enableSchemaCache' => true,
            'on afterOpen' => ['my\Log', 'handlerAfterOpen'],
],
```

### What is the expected result?

Will run my `my\Class::handlerAfterOpen`, about HumHub not my worries ..

### What do you get instead?
call_user_func() expects parameter 1 to be a valid callback, no array or string given
![image](https://user-images.githubusercontent.com/874234/31722877-ec9e101a-b425-11e7-8209-d54f29704c93.png)

### Additional info
It because after `yii\helpers\ArrayHelper::merge(config1, config2)`  get 

```
/opt/site/protected/vendor/yiisoft/yii2/base/Component.php:506:
array
  0 => string 'humhub\libs\Helpers' 
  1 => string 'SqlMode'
  2 => string 'my\Log' 
  3 => string 'handlerAfterOpen' 
```

**Comment by mikk150** [comment]
arrayhelper doesn't relly know if it's `\Callable` or not

Tho, could we do smth like

```
'on afterOpen' => [['YourAwesomeClass', 'yourAwesomeMethod'], ['TheirAwesomeClass', 'theirAwesomeMethod']
```

**Comment by rob006** [comment]
See #11269 and #13199

**Comment by samdark** [comment]
Won't be implemented in 2.0.
